### PR TITLE
Add additonal Stm32g0 socs and nucleog0b1 board.

### PR DESCRIPTION
**Pull request by str4t0m**
- dts/arm/st: g0: remove unnecessary interrupt-names from adc, and uart
- dts/arm/st: add additional g0 device definitions
- soc/arm/st_stm32: Add additional stm32g0 socs
- drivers: gpio: add definition for stm32g0 porte
- drivers: clock_control: stm32g0: add support for g051, g0b1, g0b0 socs
- board: arm: add support for the nucleo_g0b1re board
- drivers: clock_control: stm32: rm Kconfig for APB2 prescaler on g0 series
- boards: stm32g0: add jlink and stm32cubeprogrammer runners.

Flashing of stm32g0b1 currently works with stm32cubeprogrammer and jlink.
Keil has already released a dsp for the new series, but some issues lead to pyocd flashing failure,
therefore i have added stm32g071 as target for pyocd for now.

### CODE REVIEW

File: dts/arm/st/g0/stm32g070X6.dtsi [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*
```
Comment (erwango) [inline dts/arm/st/g0/stm32g070X6.dtsi:1]: I haven't seen any reference to this one (only STM32G070CB STM32G070KB STM32070RB)
Do you have any link ?

File: dts/arm/st/g0/stm32g070X8.dtsi [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/*
```
Comment (erwango) [inline dts/arm/st/g0/stm32g070X8.dtsi:1]: Same here

File: dts/arm/st/g0/stm32g071X6.dtsi [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/*
```
Comment (erwango) [inline dts/arm/st/g0/stm32g071X6.dtsi:1]: This one is not known neither

File: dts/arm/st/g0/stm32g0b0.dtsi [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+		spi3: spi@40003c00 {
+			compatible = "st,stm32-spi-fifo", "st,stm32-spi";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x40003c00 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00008000>;
+			interrupts = <25 1>;
+			status = "disabled";
+			interrupt-names = "combined";
```
Comment (erwango) [inline dts/arm/st/g0/stm32g0b0.dtsi:81]: We're not using interrupt names on spi nodes

File: dts/arm/st/g0/stm32g0b1.dtsi [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+				label = "GPIOE";
+			};
+		};
+
+		usart5: serial@40005000 {
+			compatible = "st,stm32-usart", "st,stm32-uart";
+			reg = <0x40005000 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00000100>;
+			interrupts = <29 2>;
+			interrupt-names = "combined";
```
Comment (erwango) [inline dts/arm/st/g0/stm32g0b1.dtsi:27]: Please remove interrupt-names

File: dts/arm/st/g0/stm32g0b1.dtsi [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+			status = "disabled";
+			label = "UART_5";
+		};
+
+		usart6: serial@40013c00 {
+			compatible = "st,stm32-usart", "st,stm32-uart";
+			reg = <0x40013c00 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00000200>;
+			interrupt-names = "combined";
```
Comment (erwango) [inline dts/arm/st/g0/stm32g0b1.dtsi:36]: Please remove interrupt-names

File: dts/arm/st/g0/stm32g0b1.dtsi [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+			status = "disabled";
+			label = "UART_6";
+		};
+
+		lpuart2: serial@40008400 {
+			compatible = "st,stm32-lpuart", "st,stm32-uart";
+			reg = <0x40008400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00000080>;
+			interrupts = <28 1>;
+			interrupt-names = "combined";
```
Comment (erwango) [inline dts/arm/st/g0/stm32g0b1.dtsi:47]: Please remove interrupt-names

File: dts/arm/st/g0/stm32g0b1.dtsi [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+		spi3: spi@40003c00 {
+			compatible = "st,stm32-spi-fifo", "st,stm32-spi";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x40003c00 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00008000>;
+			interrupts = <25 1>;
+			status = "disabled";
+			interrupt-names = "combined";
```
Comment (erwango) [inline dts/arm/st/g0/stm32g0b1.dtsi:91]: Please remove interrupt-names

File: dts/arm/st/g0/stm32g0b1Xb.dtsi [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <st/g0/stm32g0b1.dtsi>
+
+/ {
+	sram0: memory@20000000 {
+		reg = <0x20000000 DT_SIZE_K(144)>;
```
Comment (erwango) [inline dts/arm/st/g0/stm32g0b1Xb.dtsi:12]: 36

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by erwango** [comment]
@str4t0m Thanks for the work! Some changes required

File: dts/arm/st/g0/stm32g0.dtsi [hunk 10]
```diff
@@ -167,28 +157,101 @@
 			reg = <0x40004400 0x400>;
 			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00020000>;
 			interrupts = <28 0>;
+			interrupt-names = "combined";
```
[the remark this answers is not in the document]
Comment (str4t0m) [reply dts/arm/st/g0/stm32g0.dtsi]: is now removed.

File: dts/arm/st/g0/stm32g0.dtsi [hunk 11]
```diff
@@ -167,28 +157,101 @@
...
+			};
+		};
+
 		timers3: timers@40000400 {
 			compatible = "st,stm32-timers";
 			reg = <0x40000400 0x400>;
 			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00000002>;
 			interrupts = <16 0>;
-			interrupt-names = "global";
+			interrupt-names = "combined";
```
[the remark this answers is not in the document]
Comment (str4t0m) [reply dts/arm/st/g0/stm32g0.dtsi]: I have left it "global", as there is currently no benefit in changing it.

File: dts/arm/st/g0/stm32g0.dtsi [hunk 12]
```diff
@@ -167,28 +157,101 @@
...
 			status = "disabled";
 			label = "TIMERS_3";
 
 			pwm {
 				compatible = "st,stm32-pwm";
 				status = "disabled";
-				st,prescaler = <10000>;
+				st,prescaler = <0>;
```
[the remark this answers is not in the document]
Comment (str4t0m) [reply dts/arm/st/g0/stm32g0.dtsi:194]: This has now been moved into pr #32646

File: dts/arm/st/g0/stm32g031.dtsi [hunk 13]
```diff
@@ -1,7 +1,40 @@
 /*
  * Copyright (c) 2019 SEAL AG
+ * Copyright (c) 2020 Thomas Stranger
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #include <st/g0/stm32g0.dtsi>
```
[the remark this answers is not in the document]
Comment (str4t0m) [reply dts/arm/st/g0/stm32g031.dtsi]: done

File: dts/arm/st/g0/stm32g070X6.dtsi (hunk 1 above)
Comment (str4t0m) [reply dts/arm/st/g0/stm32g070X6.dtsi]: I have accidentally duplicated that from 71x6.

File: dts/arm/st/g0/stm32g070X8.dtsi (hunk 2 above)
Comment (str4t0m) [reply dts/arm/st/g0/stm32g070X8.dtsi]: removed

File: dts/arm/st/g0/stm32g071X6.dtsi (hunk 3 above)
Comment (str4t0m) [reply dts/arm/st/g0/stm32g071X6.dtsi:1]: It is listed in the Keil pack, but that could be an error. Should I remove it?

File: dts/arm/st/g0/stm32g0b1Xb.dtsi (hunk 9 above)
Comment (str4t0m) [reply dts/arm/st/g0/stm32g0b1Xb.dtsi:12]: The 144k should be right.

**Comment by str4t0m** [comment]
I have seen that the STM32_open_pin_data repo has been updated today (stm32cubemx 6.2.0 release).
- This is the first one that includes the definitions of the g05x socs.
- And for g0b1 it now includes the 128k flash variant:
  Additionally to the mcu/STM32G0B1R(C-E)Tx.xml they have now added a mcu/STM32G0B1R(B-C-E)Tx.xml  file...

Will we need to wait until the the pinctrl DTS files of the new release are merged in the hal before this pr can be merged, or could this pr be merged before?

File: dts/arm/st/g0/stm32g071X6.dtsi (hunk 3 above)
Comment (str4t0m) [reply dts/arm/st/g0/stm32g071X6.dtsi]: I have removed it now as I only found this in the keil dfp and not on the ST website.

**Comment by str4t0m** [comment]
I addressed all changes and rebased, @erwango could you have another look?

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**